tutorials/scripting/c_sharp/c_sharp_exports.rst

```diff
@@ -0,0 +1,354 @@
+when editing.
+	
+::
+
+	[Export(PropertyHint.ExpEasing)]
+	private float TransitionSpeed;
+
+Colors
+------
+
+Regular color given as red-green-blue-alpha value.
+
+::
+
+    [Export]
+	private Color Col;
+
+Color given as red-green-blue value (alpha will always be 1).
+
+::
+
+    [Export(PropertyHint.ColorNoAlpha)]
+	private Color Col;
+
+Nodes
+-----
+
+Nodes can't be directly exported. Instead you need to export
+a node path, then use that node path with ``GetNode()``.
+
+::
+
+	[Export]
+	private NodePath MyNodePath;
+	private Label MyNode;
+
+	public override void _Ready()
+	{   
+		MyNode = (Label)GetNode(MyNodePath);
+	}
+
+Resources
+---------
+
+::
+
+    [Export]
+	private Resource Resource;
+
+In the Inspector, you can then drag and drop a resource file
+from the FileSystem dock into the variable slot.
+
+Opening the inspector dropdown may result in an
+extremely long list of possible classes to create, however.
+Therefore, if you specify an extension of Resource such as:
+
+    [Export]
+	private AnimationNode Resource;
+
+The drop-down menu will be limited to AnimationNode and all
+its inherited classes.
+
+It must be noted that even if the script is not being run while in the
+editor, the exported properties are still editable. This can be used
+in conjunction with a :ref:`script in "tool" mode <doc_gdscript_tool_mode>`.
+
+Exporting bit flags
+-------------------
+
+Integers used as bit flags can store multiple ``true``/``false`` (boolean)
+values in one property. By using the ``Flags`` property hint, they
+can be set from the editor::
+
+    // Set any of the given flags from the editor.
+    [Export(PropertyHint.Flags, "Fire,Water,Earth,Wind")]
+	private int SpellElements = 0;
+
+You must provide a string description for each flag. In this example, ``Fire``
+has value 1, ``Water`` has value 2, ``Earth`` has value 4 and ``Wind``
+corresponds to value 8. Usually, constants should be defined accordingly (e.g.
+``priavte const int ElementWind = 8`` and so on).
+
+Export annotations are also provided for the physics and render layers defined in the project settings::
+
+	[Export(PropertyHint.Layers2dPhysics)]
+	private int Layers2dPhysics;
+	[Export(PropertyHint.Layers2dRender)]
+	private int Layers2dRender;
+	[Export(PropertyHint.Layers3dPhysics)]
+	private int layers3dPhysics;
+	[Export(PropertyHint.Layers3dRender)]
+	private int layers3dRender;
+
+Using bit flags requires some understanding of bitwise operations.
+If in doubt, use boolean variables instead.
+
+Exporting arrays
+----------------
+
+You can omit the default value, but then it would be null if not assigned.
+
+::
+
+    [Export]
+	private int[] b;
+
+Arrays with specified types which inherit from resource can be set by
+drag-and-dropping multiple files from the FileSystem dock.
+
+::
+
+	[Export]
+	private Texture[] Textures;
+	[Export]
+	private PackedScene[] Scenes;
+
+Typed arrays also work, only initialized empty:
```

Comment:
All arrays shown above and below here are typed, so it's a bit confusing. Also, we're talking about initializing them empty, but they aren't initialized in the example.

```cs
[Export]
private Vector3[] Vector3s = new Vector3[0];
[Export]
private String[] String = new String[0];
```

I would move this example above the `private int[] b` one, as the most basic one, saying they should be initialized empty (leaving out the part about types). And after that have the part about omitting default value.